Unix/provreg/provreg.c

```diff
@@ -615,7 +616,7 @@ static int _AddEntryForExtraClass(
          */
         if (strcmp(regClass->hosting, PROV_REG_HOSTING_INPROC) == 0)
         {
-            hosting = PROV_HOSTING_INPROC;
+            hosting = PROV_HOSTING_REQUESTOR;
```

Comment:
Overriding inproc silently does not sound right. You definitely need a very big error telling which provider asked for inproc, but it was made out of processes and that the provider probably needs to be updated.